### DISCUSSION TITLE
Google Assistant Integration

### DISCUSSION THREAD

**Discussion by NellyWhads** [iSoron/uhabits#577] [Feature Requests]
Planning to play with the Google assistant API to create conversational interactions that track daily activities. Could I build it as an addition to this code base? I imagine the same is possible for Siri, however, that would not be the scope of my (most likely amateurly attempted) work.

It seems it would be against one of the principles outlined in the readme: _internet connection requirement for assistant voice, and some data sharing since it parses the speech tokens._

Could I get some thoughts from the maintainers?

Stay safe!

**Comment by chennemann** [comment]
I like the idea. But you are probably right in that it would conflict with the attempt of privacy. 
A possible solution to that could be to implement/extend the API for external services such as Tasker and provide some kind of plugin that handles that. 
I don't know the status of this feature though

**Comment by iSoron** [comment]
Thank you for the suggestion, @NellyWhadsDev. Could you give some concrete examples of actions you would like Google Assistant to provide? I am not very familiar with Google Assistant integration, but from [this webpage](https://developers.google.com/assistant/app/overview) it seems to be based on the Android intent system. Would we need to include any Google SDKs during compilation to make this work?

**Comment by jonycoo** [comment]
I have found a keynote talk about that topic
https://youtu.be/vQALSeGy9aY
